### PR TITLE
adding custom prop via rest param

### PR DESCRIPTION
**Pull request by amf9t2**
### Summary
Adding custom prop functionality via rest param and refactoring to throwing error instead of warning when fit not provided. #3 

@cerner/terra

### CODE REVIEW

File: src/Arrange.jsx [hunk 1]
```diff
@@ -26,47 +14,55 @@ const propTypes = {
...
+  fitStart,
+  fill,
+  fitEnd,
+  align,
+  alignFitStart,
+  alignFill,
+  alignFitEnd,
+  ...customProps
+  }) => {
+  let alignmentFitStart = alignFitStart;
```
Comment (mhemesath) [inline src/Arrange.jsx]: do you need to re-assign these props?

Comment (amf9t2) [inline src/Arrange.jsx]: yes, to overwrite 'alignFitStart', 'alignFill', 'alignFitEnd' value with 'align' value if given.

**Comment by alex-bezek** [comment]
👍

**Comment by zhongyn** [comment]
+1

**Comment by StephenEsser** [comment]
The transpiled Arrange.js wasn't updated. You'll need to run `yarn run build` to sync the jsx and js. This is starting to happen in a lot of our components. We had talked about making it a post step in yarn, or reverting back to npm for our projects.

**Comment by amf9t2** [comment]
@StephenEsser [updated](https://github.com/cerner/terra-arrange/pull/4/commits/b7cbaebc6f485caf00a126c5106b27d04eb46f89) . Good catch.

File: src/Arrange.jsx [hunk 2]
```diff
@@ -26,47 +14,55 @@ const propTypes = {
...
   const fillClass = classNames([
     'terra-Arrange-fill',
-    { [`terra-Arrange-fill--${alignFill}`]: alignFill },
+    { [`terra-Arrange-fill--${alignmentFill}`]: alignmentFill },
   ]);
 
-  if (props.fitStart === undefined && props.fitEnd === undefined) {
-    // eslint-disable-next-line no-console
-    console.warn('At least one of the props: [fitStart, fitEnd] should be supplied.');
+  if (fitStart === undefined && fitEnd === undefined) {
```
Comment (JakeLaCombe) [inline src/Arrange.jsx]: Should there also be a check to make sure start and end are also valid?

**Pull request merged**